bob/playbook/build_internal.go

```diff
@@ -42,6 +43,8 @@ func (p *Playbook) build(ctx context.Context, task *bobtask.Task) (err error) {
 		case <-done:
 		case <-ctx.Done():
 			if errors.Is(ctx.Err(), context.Canceled) {
+				artifactId, _ := task.HashIn()
+				_ = p.localStore.ArtifactRemove(context.TODO(), artifactId.String())
```

Comment:
Guess this is the reason all artifacts are removed when hitting Ctrl-C right at the beginning, as all tasks are  flagged as canceled. I think it is not necessary to remove all artifacts in this case.

If you intended to remove a artifact which is invalid due to a canceled download you should explicitly react to the canceled download and not to a general cancel signal.

As far as i tested it, download cancel + subsequent build seems to work even with this line  commented out.

Reply:
"As far as i tested it, download cancel + subsequent build seems to work even with this line commented out."

I think the reason for this is because of the other commit "[retry download if local artifact return io.UnexpectedEOF](https://github.com/benchkram/bob/pull/227/commits/d54739f29d4293fc6fd03f6df944678f7f863add)"